### PR TITLE
GEOPY-1165: Check for properties for Property Group Type - Implement Property Group Type: Simple

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1165 - Check for properties for Property Group Type - Implement Property Group Type: Simple**
verify property_group_type when assigining children to property group

### CODE REVIEW

File: geoh5py/groups/property_group.py [hunk 1]
```diff
@@ -44,9 +44,137 @@ class GroupTypeEnum(str, Enum):
...
+        """
+        Verify that the children are of the correct type for the group type.
+
+        Raises a TypeError if the children are not of the correct type.
+
+        :param children: The children to verify.
+        :param group_type: The group type to verify against.
+        """
+        if group_type == cls.DEPTH:
+            cls._verify_depth(children)
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/groups/property_group.py:167]: created a new ptyhton file "Property Group Type" to deal with that

File: geoh5py/groups/property_group_type.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+from __future__ import annotations
+
+from abc import ABC
+from enum import Enum
+
+from ..data import Data, DataAssociationEnum, FloatData, NumericData
+
+
+class SimpleGroup(ABC):
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/groups/property_group_type.py:28]: My description is outdated. 
I made all my group inherit from Simple Group, as SimpleGroup is taking anything

**Review by domfournier — APPROVED** [review]
Second look at it I like it a lot more. Keeping the Enum let's us serialize to file much more easily. Not sure if you ended up changing your design, but if not, we can move this forward.
Well done.

File: geoh5py/shared/concatenation/property_group.py [hunk 3]
```diff
@@ -78,7 +78,11 @@ def locations(self) -> np.ndarray:
 
     @property
     def depth_(self):
-        if self.properties is None or len(self.properties) < 1:
+        if (
+            self.properties is None
+            or len(self.properties) < 1
+            or self._property_group_type != GroupTypeEnum.DEPTH
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/shared/concatenation/property_group.py:84]: This was already tackled no?

**Comment by MatthieuCMira** [comment]
> After a second look at it, I like it a lot more. Keeping the Enum let's us serialize to file much more easily. Not sure if you ended up changing your design, but if not, we can move this forward. Well done.

I don't get your commenté.
I removed the Enum as you asked.

**Comment by domfournier** [comment]
> I don't get your comment. I removed the Enum as you asked and you want it back?.

**Review by domfournier — APPROVED** [review]
This looks great. We might not even have to instantiate the PropertyGroupType since everything is static, but this works.

**Pull request merged**